src/entry/Entry.tsx

```diff
@@ -7,31 +7,48 @@ import {
   Route,
 } from 'react-router-dom';
 import { InPageNav, Loader } from 'franklin-sites';
-import useDataApi from '../utils/useDataApi';
+import { connect } from 'react-redux';
+import { Dispatch, bindActionCreators } from 'redux';
 import UniProtKBEntryConfig from '../view/uniprotkb/UniProtEntryConfig';
-import apiUrls from '../utils/apiUrls';
 import uniProtKbConverter, {
   UniProtkbUIModel,
+  UniProtkbAPIModel,
 } from '../model/uniprotkb/UniProtkbConverter';
 import { hasContent, hasExternalLinks } from '../model/utils/utils';
 import SideBarLayout from '../layout/SideBarLayout';
 import EntrySection from '../model/types/EntrySection';
 import EntryMain from './EntryMain';
 import EntryExternalLinks from './EntryExternalLinks';
+import { RootState, RootAction } from '../state/state-types';
+import * as entryActions from './state/entryActions';
 
 type MatchParams = {
   accession: string;
 };
 
-type EntryProps = RouteComponentProps<MatchParams>;
+type EntryProps = RouteComponentProps<MatchParams> & {
+  entryData: UniProtkbAPIModel | null;
+  dispatchFetchEntry: (accesion: string) => void;
+  dispatchResetEntry: () => void;
+};
+
+const Entry: React.FC<EntryProps> = ({
+  match,
+  history,
+  entryData,
+  dispatchFetchEntry,
+  dispatchResetEntry,
+}) => {
+  useEffect(() => {
+    dispatchFetchEntry(match.params.accession);
+    return function cleanup() {
```

Comment:
I think the advantage with declaring it as a non arrow function here is that we can have the word "cleanup" compared to: `return () => {...}`

Reply:
That's a very good point. It is pretty obscure otherwise what that returned fn is doing. Playing devils advocate and i'm indifferent to either of these approaches: could also do `const cleanUp = () => {...}; return cleanUp`.